test/ut3_tester/core/test_suite_builder.pkb

```diff
@@ -1114,5 +1133,305 @@ create or replace package body test_suite_builder is
+    l_annotations ut3.ut_annotations;
+  begin
+    l_annotations := ut3.ut_annotations(
+        ut3.ut_annotation(2, 'suite','testsuite', null),
+        ut3.ut_annotation(8, 'test','Some test', 'test_procedure'),
+        ut3.ut_annotation(9, 'tags',null, 'test_procedure')
+    );
+    --Act
+    l_actual := invoke_builder_for_annotations(l_annotations, 'SOME_PACKAGE');
+    --Assert
+    ut.expect(l_actual).to_be_like(
+        '%<WARNINGS>%&quot;--%tags&quot; annotation requires a tag value populated. Annotation ignored.%</WARNINGS>%'||
+        '%<TAGS/>%'
+    );
+ 
+  end;
+  
+  procedure suite_empty_tag is
+    l_actual      clob;
+    l_annotations ut3.ut_annotations;
+  begin
+      --Arrange
+    l_annotations := ut3.ut_annotations(
+        ut3.ut_annotation(2, 'suite','testsuite', null),
+        ut3.ut_annotation(3, 'tags',null, null)
+    );
+    --Act
+    l_actual := invoke_builder_for_annotations(l_annotations, 'SOME_PACKAGE');
+    --Assert
+    ut.expect(l_actual).to_be_like(
+        '%<WARNINGS><VARCHAR2>&quot;--%tags&quot; annotation requires a tag value populated. Annotation ignored, line 3</VARCHAR2></WARNINGS>%'||
+        '%<TAGS/>%'
+    );
+ 
+  end;
+
+  procedure test_duplicate_tag is
+    l_actual      clob;
+    l_annotations ut3.ut_annotations;
+  begin
+      --Arrange
+    l_annotations := ut3.ut_annotations(
+        ut3.ut_annotation(2, 'suite','testsuite', null),
+        ut3.ut_annotation(8, 'test','Some test', 'test_procedure'),
+        ut3.ut_annotation(9, 'tags','testtag,testtag1,testtag', 'test_procedure'),
+        ut3.ut_annotation(10, 'tags',' testtag,testtag1,testtag2', 'test_procedure')
+    );
+    --Act
+    l_actual := invoke_builder_for_annotations(l_annotations, 'SOME_PACKAGE');
+    --Assert
+    ut.expect(l_actual).to_be_like(
+        '%<UT_SUITE_ITEM>' ||
+        '%<NAME>test_procedure</NAME><DESCRIPTION>Some test</DESCRIPTION><PATH>some_package.test_procedure</PATH>' ||
+        '%<TAGS><VARCHAR2>testtag</VARCHAR2><VARCHAR2>testtag1</VARCHAR2><VARCHAR2>testtag2</VARCHAR2></TAGS>%'||
+        '%</UT_SUITE_ITEM>%'
+    );
+ 
+  end;
+  
+  procedure suite_duplicate_tag is
+    l_actual      clob;
+    l_annotations ut3.ut_annotations;
+  begin
+      --Arrange
+    l_annotations := ut3.ut_annotations(
+        ut3.ut_annotation(2, 'suite','testsuite', null),
+        ut3.ut_annotation(3, 'tags','suitetag,suitetag1,suitetag', null),
+        ut3.ut_annotation(4, 'tags',' suitetag1,suitetag2', null)
+    );
+    --Act
+    l_actual := invoke_builder_for_annotations(l_annotations, 'SOME_PACKAGE');
+    --Assert
+    ut.expect(l_actual).to_be_like(
+        '%<UT_LOGICAL_SUITE>' ||
+        '%<NAME>some_package</NAME><DESCRIPTION>testsuite</DESCRIPTION><PATH>some_package</PATH>' ||
+        '%<TAGS><VARCHAR2>suitetag</VARCHAR2><VARCHAR2>suitetag1</VARCHAR2><VARCHAR2>suitetag2</VARCHAR2></TAGS>%'||
+        '%</UT_LOGICAL_SUITE>%'
+    );
+ 
+  end;
+
+  procedure test_empty_tag_between is
```

Comment:
What about tag that has spaces inside?
`--%tags( my first tag, my second tag, some other tag )`
How will that work?
Are spaces in tags allowed?
Is comma the only disallowed character?

Reply:
The spaces are not allowed it will ignore such tag.